src/Symfony/Component/HttpClient/Response/HttplugPromise.php

```diff
@@ -50,7 +50,7 @@ public function getState(): string
     /**
      * {@inheritdoc}
      *
-     * @return Psr7ResponseInterface|mixed
+     * @return Psr7ResponseInterface
```

Comment:
I discussed about this one with @Nyholm. This interface is `@return mixed` but it looks like we only technically return `Psr7ResponseInterface` in our implementation.

Reply:
since the class is internal and final, we can then turn this into a real return-type, that would allow verifying the assertion a but thanks to tests

Reply:
tests caught than this can return null, when $unwrap = false